### DISCUSSION TITLE
Middleware vs Prefetch

### DISCUSSION THREAD

**Discussion by paqstd-dev** [vercel/next.js#50628] [App Router] [answered]
Can someone explain to me why middleware works on prefetch requests? How can I disable this behavior?

For example, I would like to specify a specific cookie depending on the current language. There is no built-in ability to process languages and translations for app router.

**Comment by jvandenaardweg** [comment, marked as answer]
You can disable the Middleware for certain request headers, including the prefetch requests. Take a look here: https://github.com/vercel/next.js/discussions/37736#discussioncomment-7274460
